src/ui_basic/panel.h

```diff
@@ -347,6 +349,11 @@ class Panel : public boost::signals2::trackable {
 
 	const PanelStyle panel_style_;
 
+	// Never call this function, except when you need Widelands to stay responsive
+	// during a costly operation and you can guarantee that it will not interfere
+	// with the "normal" graphics refreshing done periodically from `Panel::do_run`.
+	void do_redraw_now();
```

Comment:
As long as Widelands is single-threaded, such "interfering" is actually not possible since redraws will be carried out smoothly one after the other; calling this function could not have any bad side effects except that there is no need for it normally. When Widelands becomes multithreaded, synchronous drawing orders from multiple threads would be **very** bad, but I already have asserts in the multithreading branch that will catch such cases.